### PR TITLE
Disable analytics if query params or token exist in url

### PR DESCRIPTION
**Pull request by adimiz1**
### Brief Summary of Changes
Disable addition of analytics string if query parameters or auth token already exist on the url  

#### What does this PR address?
- [ ] GitHub issue (Add reference - #XX)
- [ ] Refactoring
- [ ] New feature
- [x] Bug fix
- [x] Adds more tests

#### Are tests included?
- [x] Yes
- [ ] No

#### Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I ran the full test suite before pushing the changes and all the tests pass.

### CODE REVIEW

File: url-gen/src/main/kotlin/com/cloudinary/asset/Asset.kt [hunk 1]
```diff
@@ -162,7 +162,11 @@ abstract class BaseAsset constructor(
                 mutableSource
             ).joinToString("/").cldMergeSlashedInUrl()
 
-        val analytics = if (urlConfig.analytics) "_a=${generateAnalyticsSignature()}" else null
+        var urlObject = URL(url)
+        val analytics = if (urlConfig.analytics && cloudConfig.authToken == null && urlObject.query == null)
+            "_a=${generateAnalyticsSignature()}"
+        else null
```
Comment (const-cloudinary) [inline url-gen/src/main/kotlin/com/cloudinary/asset/Asset.kt:168]: indentation is a bit unclear..., if the condition becomes too long maybe switch to classic `if else`, or extract the condition to a variable

File: url-gen/src/test/kotlin/com/cloudinary/AuthTokenTest.kt [hunk 2]
```diff
@@ -165,7 +164,7 @@ class AuthTokenTest {
         val url =
             cloudinary.image().generate("sample.jpg")
         assertEquals(
-            "https://res.cloudinary.com/test123/image/upload/sample.jpg?__cld_token__=st=11111111~exp=11111411~hmac=ec2d0e9a87f6dd5147c36f39c5e88c28fdaa53d1e04f3d70160f7ebe62a60c7a&_a=$expectedAnalytics",
+            "https://res.cloudinary.com/test123/image/upload/sample.jpg?__cld_token__=st=11111111~exp=11111411~hmac=ec2d0e9a87f6dd5147c36f39c5e88c28fdaa53d1e04f3d70160f7ebe62a60c7a",
```
Comment (const-cloudinary) [inline url-gen/src/test/kotlin/com/cloudinary/AuthTokenTest.kt:167]: test name `testAuthTokenWithAnalytics` contradicts assertion being made

I would keep the test and add another one to cover the new use-case.

Comment (adimiz1) [reply url-gen/src/test/kotlin/com/cloudinary/AuthTokenTest.kt:167]: Not sure what you mean
We cant keep the current test as it'll fail , when auth token exist analytics will not appear on the URL
So I changed the current test to work with `analytics = true` but the assert is to make sure that the analytics string does not appear

File: url-gen/src/main/kotlin/com/cloudinary/asset/Asset.kt (hunk 1 above)
Comment (adimiz1) [reply url-gen/src/main/kotlin/com/cloudinary/asset/Asset.kt:168]: Done

**Review by patrick-tolosa — APPROVED** [review]

File: url-gen/src/test/kotlin/com/cloudinary/AuthTokenTest.kt (hunk 2 above)
Comment (patrick-tolosa) [reply url-gen/src/test/kotlin/com/cloudinary/AuthTokenTest.kt:167]: I think the idea is that the test tests the opposite

It's not "AuthTokenWithAnalytics" but rather "DisableAnalyticsWithAuthToken"

**Review by const-cloudinary — APPROVED** [review]

**Pull request merged**